### PR TITLE
support for new node feature next-nonce of release 6.2.0

### LINKED ISSUE #1288: make integration tests in CI deterministic
currently some integration tests always randomly fail

### PR DESCRIPTION
**Pull request by subhod-i**
Fix #1288 along with #1260

Node API `next-nonce` returns 404 if an account doesn't exist in the state tree, Learn more: https://github.com/aeternity/aeternity/blob/684e89ffe45e3f15a7e50bb71773951a4a9d9b83/apps/aecore/src/aec_next_nonce.erl#L8

### CODE REVIEW

File: src/tx/tx.js [hunk 1]
```diff
@@ -481,13 +481,19 @@ async function getAccountNonce (accountId, nonce) {
  * @param {Object} params Object which contains all tx data
  * @return {Object} { ttl, nonce, fee } Object with account nonce, absolute ttl and transaction fee
  */
-async function prepareTxParams (txType, { senderId, nonce: n, ttl: t, fee: f, gas, absoluteTtl, vsn }) {
+async function prepareTxParams (txType, { senderId, nonce: n, ttl: t, fee: f, gas, absoluteTtl, vsn, strategy }) {
   const account = await this.getAccount(senderId).catch(e => ({ nonce: 0 }))
   // Is GA account
   if (account.contractId) {
     n = 0
```
Comment (davidyuk) [inline src/tx/tx.js:488]: As I see, `getAccountNextNonce` always returns 0 for generalized accounts, so maybe we can get rid of `getAccount` request?

Comment (subhod-i) [reply src/tx/tx.js:488]: Good. let me optimize the method again

File: src/tx/tx.js [hunk 2]
```diff
@@ -481,13 +481,19 @@ async function getAccountNonce (accountId, nonce) {
...
   } else {
-    n = n || (account.nonce + 1)
+    if (!n) {
+      try {
+        n = (await this.api.getAccountNextNonce(senderId, { strategy: 'continuity' })).nextNonce
```
[the remark this answers is not in the document]
Comment (subhod-i) [reply src/tx/tx.js]: This history I made just to see if the tests can pass with `continuity`

**Comment by subhod-i** [comment]
@davidyuk  Thanks for the input. Method `prepareTxParams` updated incorporating all the commets.

File: src/tx/tx.js [hunk 3]
```diff
@@ -481,14 +481,9 @@ async function getAccountNonce (accountId, nonce) {
...
-  const account = await this.getAccount(senderId).catch(e => ({ nonce: 0 }))
-  // Is GA account
-  if (account.contractId) {
-    n = 0
-  } else {
-    n = n || (account.nonce + 1)
-  }
+async function prepareTxParams (txType, { senderId, nonce: n, ttl: t, fee: f, gas, absoluteTtl, vsn, strategy }) {
+  const { nextNonce } = await this.api.getAccountNextNonce(senderId, { strategy }).catch(e => ({ nextNonce: 1 }))
+  n = n || nextNonce
```
Comment (davidyuk) [inline src/tx/tx.js:486]: As it was in Oracles PR: let's don't request nonce if it was passed
```suggestion
  n = n || (await this.api.getAccountNextNonce(senderId, { strategy }).catch(e => ({ nextNonce: 1 }))).nextNonce
```
this line gets too long, not sure how to handle it

**Review by davidyuk — APPROVED** [review]
fine to merge after fixing the last

**Pull request merged**